CRM/Core/Permission/Joomla.php

```diff
@@ -131,4 +135,74 @@ public function checkGroupRole($array) {
     return FALSE;
   }
 
+  /**
+   * @inheritDoc
+   */
+  public function upgradePermissions($permissions) {
```

Comment:
Sounds fine. I'm not aware if variable names generate strict warnings about signatures. In other, strongly-typed, languages, variable names are not part of signatures.